src/providers/user.provider.tsx

```diff
@@ -22,26 +36,69 @@ export interface IProps {
 
 const UserProvider = ({children}: IProps) => {
   const [user, setUser] = useState<Patient | undefined>(undefined)
-  const [bloodPressures, setBloodPressures] = useState<BloodPressure[]>([])
+  const [bloodPressures, setBloodPressures] = useState<
+    BloodPressure[] | undefined
+  >(undefined)
+
+  const [medications, setMedications] = useState<Medication[] | undefined>(
+    undefined,
+  )
 
   const updatePatientData = async (
     patientResponseData: PatientResponseData,
   ) => {
-    // Todo - write/update patient offline
-    // Todo - write/update blood pressures offline
     const {
       patient_id,
       full_name,
       password_digest,
       blood_pressures,
+      medications,
     } = patientResponseData
-    setUser({patient_id, full_name, password_digest})
-    setBloodPressures(blood_pressures.slice())
+    const userData = {patient_id, full_name, password_digest}
+    const bloodPressuresData = blood_pressures.slice()
+    const medicationsData = (medications ?? []).slice()
```

Comment:
What are you slicing exactly?

Reply:
Slice is going to make a copy of the array from the api json. To avoid creating a reference to the original array which I want to avoid. Basically like a redux pure function - to ensure that there will be no side effects, i.e., it does not alter any of  original input data

Reply:
I would just create a copy with `[...medications]`

Reply:
Good suggestion. Updated/changed.